npm/vite-dev-server/test/resolveConfig.spec.ts

```diff
@@ -21,80 +23,83 @@ const getViteDevServerConfig = (projectRoot: string) => {
     framework: 'react',
   } as unknown as ViteDevServerConfig
 }
+const MAJOR_VERSIONS: [4, 5] = [4, 5]
 
 describe('resolveConfig', function () {
   this.timeout(1000 * 60)
 
-  context('config resolution', () => {
-    it('with <project-root>/vite.config.js', async () => {
-      const projectRoot = await scaffoldSystemTestProject('vite-inspect')
-      const viteDevServerConfig = getViteDevServerConfig(projectRoot)
+  MAJOR_VERSIONS.forEach((version) => {
+    context(`config resolution: v${version}`, () => {
+      it('with <project-root>/vite.config.js', async () => {
+        const projectRoot = await scaffoldSystemTestProject(`vite${version}-inspect`)
+        const viteDevServerConfig = getViteDevServerConfig(projectRoot)
 
-      const viteConfig = await createViteDevServerConfig(viteDevServerConfig, vite)
+        const viteConfig = await createViteDevServerConfig(viteDevServerConfig, version === 5 ? vite5 : (vite4 as unknown as Vite))
```

Comment:
Rather than flexing this everywhere. You could potentially make MAJOR_VERSIONS an array of objects like:

```
[
  {
    version: 4,
    vite: vite4
  },
  {
    version: 5,
    vite: vite5
  }
]
```

Reply:
The issue right now is we implicitly support versions below 4 as well, including 2 and 3 since we have not deprecated them. But for a unit test this probably isn't a big deal so I'll get it updated